plugins/modem-manager/fu-mm-device.c

```diff
@@ -239,68 +384,210 @@ fu_mm_device_detach_fastboot (FuDevice *device, GError **error)
 }
 
 static gboolean
-fu_mm_device_inhibit (FuMmDevice *self, GError **error)
+fu_mm_device_detach (FuDevice *device, GError **error)
 {
-	MMModem *modem = mm_object_peek_modem (self->omodem);
+	g_autoptr(FuDeviceLocker) locker  = NULL;
+	guint32 current_method;
 
-	/* prevent NM from activating the modem */
-	g_debug ("inhibit %s", mm_modem_get_device (modem));
-	if (!mm_manager_inhibit_device_sync (self->manager,
-					     mm_modem_get_device (modem),
-					     NULL, error))
+	locker = fu_device_locker_new (device, error);
+	if (locker == NULL)
 		return FALSE;
 
-	/* success: the device will disappear */
-	return TRUE;
+	current_method = fu_device_get_current_method (device);
+
+	/* last cycle before considering operation finished */
+	if (current_method == 0)
+		return TRUE;
+
+	/* fastboot */
+	if (current_method == FU_MM_DEVICE_SUPPORTED_METHOD_FASTBOOT)
+		return fu_mm_device_detach_fastboot (device, error);
+
+	/* qmi pdc doesn't require any detach */
+	if (current_method == FU_MM_DEVICE_SUPPORTED_METHOD_QMI_PDC)
+		return TRUE;
+
+	/* should not get here */
+	g_set_error_literal (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "unexpected detach method requested");
+	return FALSE;
+}
+
+typedef struct {
+	gchar  *filename;
+	GBytes *bytes;
+} FileInfo;
+
+typedef struct {
+	FuMmDevice	*device;
+	GError		*error;
+	GList		*file_infos;
+	gsize		 total_written;
+	gsize		 total_bytes;
+} ArchiveIterateContext;
```

Comment:
FuMmArchiveIterateCtx perhaps?

Reply:
pushed a new fixup to handle this renaming